From a056a039d37ba93b1a3859da28156f86fc914090 Mon Sep 17 00:00:00 2001
From: wu-changsheng <wuchangsheng2@huawei.com>
Date: Sat, 3 Dec 2022 22:57:13 +0800
Subject: [PATCH] expand rxtx mbuf pool

---
 src/lstack/include/lstack_dpdk.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lstack/include/lstack_dpdk.h b/src/lstack/include/lstack_dpdk.h
index c042ef5..54b8fc2 100644
--- a/src/lstack/include/lstack_dpdk.h
+++ b/src/lstack/include/lstack_dpdk.h
@@ -15,7 +15,7 @@
 
 #include "gazelle_opt.h"
 
-#define RXTX_NB_MBUF        (256 * 2000) /* mbuf per connect * connect num. size of mbuf is 2536 Byte */
+#define RXTX_NB_MBUF        (512 * 2000) /* mbuf per connect * connect num. size of mbuf is 2536 Byte */
 #define RXTX_CACHE_SZ       (VDEV_RX_QUEUE_SZ)
 #define KNI_NB_MBUF         (DEFAULT_RING_SIZE << 4)
 
-- 
2.23.0

