From eb3f797f679605c26355aea152b249dfd3852786 Mon Sep 17 00:00:00 2001
From: kircher <majun65@huawei.com>
Date: Sun, 29 Jan 2023 21:02:33 +0800
Subject: [PATCH] add log message when wait for connecting to ltran

---
 src/lstack/core/lstack_control_plane.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lstack/core/lstack_control_plane.c b/src/lstack/core/lstack_control_plane.c
index bf55df5..bc2c55d 100644
--- a/src/lstack/core/lstack_control_plane.c
+++ b/src/lstack/core/lstack_control_plane.c
@@ -97,6 +97,7 @@ static int32_t connect_to_ltran(int32_t times, int32_t interval)
             return sockfd;
         }
 
+        LSTACK_LOG(INFO, LSTACK, "wait for connecting to ltran\n");
         int32_t ret = posix_api->connect_fn(sockfd, (struct sockaddr*)&address, sizeof(address));
         if (ret == 0) {
             return sockfd;
-- 
2.33.0

